scripts/plotting/time_series.py

```diff
@@ -0,0 +1,630 @@
+                        yrs_s = []
+                        #Grab first year avialable for December
+                        for val in arr.time[1:]:
+                            yr = val.item().timetuple().tm_year
+                            #Need to put year in 4-digit expression
+                            #so grab leading zeros (ie yr 31 would be 0031)
+                            yrs_s.append(str(yr).zfill(4))
+                        yrs_s = np.unique(yrs_s)
+                        yrs[case_name] = yrs_s
+
+                        yrs_mean = [arr.sel(time=i).mean().values for i in yrs[case_name]]
+                        vals[var][case_name][season] = yrs_mean
+
+                    else:
+                        #Get years
+                        yrs_s = []
+                        for val in arr.time:
+                            yr = val.item().timetuple().tm_year
+                            yrs_s.append(str(yr).zfill(4))
+                        yrs_s = np.unique(yrs_s)
+                        yrs[case_name] = yrs_s
+
+                        yrs_mean = [arr.sel(time=i).mean().values for i in yrs[case_name]]
+                        vals[var][case_name][season] = yrs_mean
+
+    return vals, yrs, del_s, units
+
+########
+
+def _set_ymargin(ax, top, bottom):
+    """
+    Allow for custom padding of plot lines and axes borders
+    -----
+    """
+    ax.set_ymargin(0)
+    ax.autoscale_view()
+    lim = ax.get_ylim()
+    delta = np.diff(lim)
+    top = lim[1] + delta*top
+    bottom = lim[0] - delta*bottom
+    ax.set_ylim(bottom,top)
+
+########
+
+def _format_yaxis(ax, case_num, unit, **kwargs):
+    """
+    Gather variable data and format y-axis
+    -----
+        - Set the y-axis plot limits to guarantee data range from all cases (including baseline)
+        - Pad the top of plot to allow for flexible-sized legend in top left corner
+            -> For multi-case, this will pad the plot according to number of cases
+    """
+
+    #Set up plot details, if applicable from the adf_variable_defaults.yaml file
+    if 'ts' in kwargs:
+        if "units" in kwargs['ts']:
+            print("Looks like desired units are different than from raw file...\n")
+            unit = kwargs['ts']["units"]
+
+    ax.set_ylabel(unit,fontsize=20,labelpad=12)
+
+    #Attempt flexible pad based on number of cases for both single and
+    #multi-case scenarios, too
+    pad = 0.075*case_num
+    _set_ymargin(ax, top=pad, bottom=0.1)
+
+    return ax
+
+########
+
+def _format_xaxis(ax, yrs):
+    """
+    Gather climo year data and format x-axis
+    -----
+        - Set the x-axis plot limits to guarantee data range from all cases (including baseline)
+        - Set minor and major locators based on number of years
+        - Round the range to the nearest 5-year interval for cleaner appearance
+    """
+
+    #Grab all unique years and find min/max years
+    uniq_yrs = sorted({x for v in yrs.values() for x in v})
```

Comment:
I think (?) you can remove the curly brackets here:

```suggestion
    uniq_yrs = sorted(x for v in yrs.values() for x in v)
```

Reply:
You are correct, doesn't seem to be necesary. Thanks